src/main/java/com/azure/cosmos/examples/changefeed/SampleChangeFeedEstimator.java

```diff
@@ -0,0 +1,217 @@
+                              .delayElement(Duration.ofSeconds(10))
+                              .flatMap(value -> changeFeedProcessorMainInstance
+                                  .stop()
+                                  .subscribeOn(Schedulers.boundedElastic())
+                                  .timeout(Duration.ofSeconds(10))))
+                    .subscribe();
+            } catch (Exception ex) {
+                logger.error("Change feed processor did not start and stopped in the expected time", ex);
+                throw ex;
+            }
+
+            Thread.sleep(Duration.ofSeconds(20).toMillis());
+
+            ChangeFeedProcessor changeFeedProcessorSideCart = new ChangeFeedProcessorBuilder()
+                .hostName("side-cart")
+                .feedContainer(feedContainer)
+                .leaseContainer(leaseContainer)
+                .handleChanges(nodes -> {
+                    logger.error("This never needs to be called, as this is just monitoring the state");
+                })
+                .buildChangeFeedProcessor();
+
+            AtomicInteger totalLag = new AtomicInteger();
+            // <EstimatedLag>
+            Mono<List<ChangeFeedProcessorState>> currentState = changeFeedProcessorMainInstance.getCurrentState();
+            currentState.map(state -> {
+                for (ChangeFeedProcessorState changeFeedProcessorState : state) {
+                    totalLag.addAndGet(changeFeedProcessorState.getEstimatedLag());
+                }
+                return state;
+            }).subscribe();
+
+            // Initially totalLag should be zero
+            logger.info("Initially total lag is : {}", totalLag.get());
+
+            totalLag.set(0);
+            currentState = changeFeedProcessorSideCart.getCurrentState();
+            currentState.map(state -> {
+                for (ChangeFeedProcessorState changeFeedProcessorState : state) {
+                    totalLag.addAndGet(changeFeedProcessorState.getEstimatedLag());
+                }
+                return state;
+            }).subscribe();
+
+            // Initially totalLag should be zero
+            logger.info("Initially total lag is : {}", totalLag.get());
+            // </EstimatedLag>
+
+            //These two lines model an application which is inserting ten documents into the feed container
+            logger.info("Start application that inserts documents into feed container");
+            createNewDocumentsCustomPOJO(feedContainer, 10);
+
+            currentState = changeFeedProcessorMainInstance.getCurrentState();
+            currentState.map(state -> {
+                for (ChangeFeedProcessorState changeFeedProcessorState : state) {
+                    totalLag.addAndGet(changeFeedProcessorState.getEstimatedLag());
+                }
+                return state;
+            }).block();
+
+            // Finally, totalLag should be equal to number of documents created
```

Comment:
In practice we cannot guarantee the equality here. We should change this instead to say that the "totalLag should be greater or equal than number of documents created".